config/core/test-configs.yaml

```diff
@@ -2539,6 +2546,7 @@ test_configs:
       - kselftest-cpufreq
       - kselftest-futex
       - kselftest-kvm
+      - kselftest-landlock
```

Comment:
Looks like there's some problem:
https://storage.staging.kernelci.org/kernelci/staging-mainline/staging-mainline-20220824.1/arm64/defconfig+kselftest/gcc-10/lab-broonie/kselftest-landlock-meson-gxl-s905x-libretech-cc.html#L1802
The `landlock` collection is missing from the kselftest tarball, so there's no results.

Also the jobs are all failing now, they could complete until the end of August.

Reply:
There's a landlock build issue:
```
make[3]: Entering directory '/tmp/kci/linux/tools/testing/selftests/landlock'
make[3]: *** No rule to make target '/tmp/kci/linux/build/kselftest/landlock/base_test', needed by 'all'.  Stop.
make[3]: Leaving directory '/tmp/kci/linux/tools/testing/selftests/landlock'
```

Reply:
It looks like the job failures aren't specific to landlock - it looks like all arm64 boots with the kselftest custom kernel broke in the same way starting at the same time. The build issue is still a problem of course.

Reply:
> There's a landlock build issue:
> 
> ```
> make[3]: Entering directory '/tmp/kci/linux/tools/testing/selftests/landlock'
> make[3]: *** No rule to make target '/tmp/kci/linux/build/kselftest/landlock/base_test', needed by 'all'.  Stop.
> make[3]: Leaving directory '/tmp/kci/linux/tools/testing/selftests/landlock'
> ```

I still cannot reproduce this build issue. Where can we find these logs and the related artifacts?

Reply:
Not the same job but the error showing up in:

https://storage.kernelci.org/next/master/next-20220908/arm64/defconfig/gcc-10/logs/kselftest.log

is

make[3]: Entering directory '/tmp/kci/linux/tools/testing/selftests/landlock'
make[3]: *** No rule to make target '/tmp/kci/linux/build/kselftest/landlock/base_test', needed by 'all'.  Stop.
make[3]: Leaving directory '/tmp/kci/linux/tools/testing/selftests/landlock'

Reply:
Poking at things a bit it seems to be an effect caused by building from the toplevel target, building directly in the kselftest directory seems fine. I suspect the explicit dependencies on the headers in the Makefile are triggering things, didn't fully investigate though.

Reply:
`cd /tmp && make TARGETS=landlock O=/tmp/build -C/tmp/linux kselftest-gen_tar` works fine for me. :thinking:

Reply:
I successfully reproduced it with one of the latest version, I'm looking into it.

Reply:
FYI, `$(khdr_dir)/linux/landlock.h` is the issue when building out of tree. I'll send a patch to fix that and handle all build cases. In the meantime, just removing this dependency should do the trick.

Reply:
Yes it's an issue with the fixes to be able to build from the top-level directory as the landlock Makefile is doing something special with header files etc.  There's a mailing list discussion about this:
https://lore.kernel.org/lkml/a459363217b1847c0f206a5dbdf181cb21cf3d0c.1659557290.git.guillaume.tucker@collabora.com/
So I think it would be good to first fix the build, I can add this patch on staging to see if that fixes things and whether the tests actually run once they're built.

Reply:
Actually that patch is already in linux-next, so there's another (related) problem here.  Still I think ideally the way out of this is to fix the kselftest landlock build.

Reply:
This fixes the issue:
```diff
--- a/tools/testing/selftests/landlock/Makefile
+++ b/tools/testing/selftests/landlock/Makefile
@@ -17,5 +17,5 @@ khdr_dir = $(top_srcdir)/usr/include
 $(OUTPUT)/true: true.c
        $(LINK.c) $< $(LDLIBS) -o $@ -static

-$(OUTPUT)/%_test: %_test.c $(khdr_dir)/linux/landlock.h ../kselftest_harness.h common.h
+$(OUTPUT)/%_test: %_test.c ../kselftest_harness.h common.h
        $(LINK.c) $< $(LDLIBS) -o $@ -lcap -I$(khdr_dir)

```

I'm working on a better patch though.

Reply:
I'll Cc you @gctucker ;)

Reply:
The fix is now in linux-next with commit [a52540522c95 ("selftests/landlock: Fix out-of-tree builds")](https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?id=4908e7a3045c409fd31c4cb1f8224457dbe87bad).

Reply:
Great, we'll be able to confirm whether that worked with the staging linux-next job.

Reply:
Can you confirm that it works @gctucker? Getting your Tested-by tag would help: https://lore.kernel.org/linux-kselftest/2879477f-82d9-9d39-13b6-9cc60a3f14c7@digikod.net/

Reply:
Lots of platform failed to boot due to a bug in linux-next, I think that's now fixed so let's wait for today's results.

Reply:
Still no results as the boards that are supposed to run the test failed to boot...
https://staging.kernelci.org/test/job/kernelci/branch/staging-next/kernel/staging-next-20220923.0/plan/kselftest-landlock/

Reply:
Ah it looks like a known issue where some debug network interfaces are brought up as a side-effect of enabling some configs with kselftest builds:
https://storage.staging.kernelci.org/kernelci/staging-next/staging-next-20220923.0/arm64/defconfig+kselftest/gcc-10/lab-baylibre/kselftest-landlock-meson-gxl-s905x-libretech-cc.html#L876

The landlock tests were built fine though:
https://storage.staging.kernelci.org/kernelci/staging-next/staging-next-20220923.0/arm64/defconfig+kselftest/gcc-10/logs/kselftest.log

Reply:
The fix is now in Linux 6.0-rc7 .